module4/interview_resources/technical_interview_questions.md

```diff
@@ -0,0 +1,144 @@
+</section>
+
+## Questions
+
+### Fullstack Development Questions
+
+* Define the term ‘MVC’ and explain how an application is architected when following MVC patterns.
+* Describe each of the four fundamental concepts of object-oriented programming. How would you implement each of these in your code?
+* What can you tell me about RESTful APIs?
+* What are some tools and strategies you use to prevent shipping unstable code to production?
+* What factors influence whether you’ll take a progressive enhancement vs. graceful degradation approach to building an application?*
+* Describe the purpose of a JSON Web Token*
+* How does JSON work?
+* What are libraries you have used?
+* What are frameworks you have used?
+* What is CORS?
+* In as much detail as possible, describe the HTTP request-response cycle.
+* What does a 200 status code mean?
+* What does a 400 status code mean?
+* Why use cookies?*
+* What’s the difference between a session and a cookie?
+* Have you used local storage?
+* Explain how a GET request works.
+* What is TDD and what benefits does it provide? 
+* What is the difference between unit, integration, and acceptance-level testing? Can you talk about times when you've done either?
+* Tell me two advantages of testing your code.
+* What is your code review process?
+* What are the benefits of using a version control system like Git?
+* How do you resolve a merge conflict?
+* Explain how a branch works in git.
+* How do you make sure your project meets the requirements you have received?
+* What IDE do you use?
+* Do you know the concept of SCRUM?*
+* Explain what Agile is.
+* What is HTTP and give a brief definition?
+* Explain a git merge workflow vs a git rebase workflow
+* What's a background worker? When would we want to use a background worker?*
+* What’s the difference between authentication and authorization? What tools have you used to set up either of these in the past?
+* What does API stand for and how do you define it?
+* What does it mean to compile code?*
+* Tell me two advantages of testing your code.
+* My production and development environments are using large sets of data. What are some ways to optimize my tests to run quickly, but be thorough?
+* What is a Mock vs. a Stub in testing? When would you use each?
+* In as much detail as possible, describe DNS.*
+* Describe what an SOA (Service Oriented Archicture) application is and looks like? What are the benefits and drawbacks?
+* Explain to me the general meaning behind each series of error codes (ex. 200's, 300's, 400's, 500's).
+* Describe how you've built error-handling into one of your applications in order to gracefully handle exceptions. 
+
+
+
+### FE-specific Technical Questions
```

Comment:
Now that we have these sections and a general fullstack section, it makes it seem like there are lots more FE questions than BE questions. Are all these FE questions still relevant? It might also be worth thinking about whether all these questions are good interview study questions or whether they're just quick recall questions that are more flashcard-esque and not going to be used in a simulation/interview
@kaylagordon @hfaerber

Reply:
Yes I agree. These haven't been updated in quite a long time. As for the interview simulation, questions will be geared toward their chosen role so if someone chooses a FE or BE leaning role, the older questions would still be good for practice I think. We can/should definitely review for relevance and filling in any additional questions that feel like a good fit for practicing.